app/models/access.rb

```diff
@@ -0,0 +1,102 @@
+class Access < ApplicationRecord
+  ALLOWED_RESOURCE_TYPES = %w[Organization FacilityGroup Facility].freeze
+
+  belongs_to :user
+  belongs_to :resource, polymorphic: true, optional: true
+
+  enum role: {
+    super_admin: "super_admin",
+    admin: "admin",
+    analyst: "analyst"
+  }
+
+  validates :role, presence: true
+  validates :resource_type, inclusion: {in: ALLOWED_RESOURCE_TYPES}, allow_nil: true
+
+  class << self
+    def can_manage_facilities?(record)
+      return true if super_admin.exists?
+
+      manageable_facilities = facilities(:manage)
+
+      if record.instance_of?(Facility)
+        manageable_facilities.where(id: record).exists?
+      else
+        manageable_facilities.exists?
+      end
+    end
+
+    def can_manage_facility_groups?(record)
+      return true if super_admin.exists?
+
+      manageable_facility_groups = facility_groups(:manage)
+
+      if record.instance_of?(FacilityGroup)
+        manageable_facility_groups.where(id: record).exists?
+      else
+        manageable_facility_groups.exists?
+      end
+    end
+
+    def can_manage_organizations?(record)
+      return true if super_admin.exists?
+
+      manageable_organizations = organizations(:manage)
+
+      if record.instance_of?(Organization)
+        manageable_organizations.where(id: record).exists?
+      else
+        manageable_organizations.exists?
```

Comment:
What else would we pass in except an instance of an Org?  

The only case not covered by these `can_` methods here would be a brand new Simple.org setup, but I think that would be a super admin type thing to create new orgs to get things up and running.  

```
if admin.can_add_new_orgs?
  <%= form_for Organization.new %>
```

and then I think:

```
def can_add_new_orgs?
  super_admin?
end
```